test/framework/e2e_framework.go

```diff
@@ -134,8 +134,10 @@ func generateCassOperatorKustomization(namespace string) error {
 kind: Kustomization
 
 resources:
-- github.com/k8ssandra/cass-operator/config/default?ref=v1.9.0
-namespace: {{ .Namespace }}
+- github.com/k8ssandra/cass-operator/config/deployments/cluster?ref=6a78a8237173d9322e6c0cab94c615b1f043a906
```

Comment:
This isn't something to fix in this PR, but shouldn't we make these versions configured to just one place? Like the Makefile and when you run the make <any-test-scenario>, it would then modify these Kustomization files? I know kustomize doesn't allow remote-resource refs to be set on the higher level, but the way we now define and configure these is quite bad.

Reply:
And I guess this applies to the whole `generateCassOperatorKustomization` function. E2E tests shouldn't have a difference in how they deploy these components compared to what we provide to our users. While we hope to maintain that they don't differ or that these patches are not necessary - our e2e tests can't actually verify that's the case. In worst case, the namespace selectors here fix a bug that was introduced in the real kustomization rules (it wouldn't be the first time kustomization rules behave unlike we expected them to)

Reply:
>This isn't something to fix in this PR, but shouldn't we make these versions configured to just one place?

Agree with both points. Definitely an area for improvement and for a different PR.

`generateCassOperatorKustomization` is no longer used. I will only open a separate PR to remove it. `generateK8ssandraOperatorKustomization` is used in the same way that users would. It simply adds another kustomization layer for setting the namespace.

Reply:
I don't quite follow this discussion, but if it isn't something to fix in this PR I think I'm safe to mark these comments as resolved?